telegram/src/telegram/bot/client/client_impl.hpp

```diff
@@ -0,0 +1,120 @@
+      const CopyMessageMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+  
+  ForwardMessageRequest ForwardMessage(
+      const ForwardMessageMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  GetChatRequest GetChat(
+      const GetChatMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  GetFileRequest GetFile(
+      const GetFileMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  GetMeRequest GetMe(const RequestOptions& request_options) override;
+
+  GetUpdatesRequest GetUpdates(
+      const GetUpdatesMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  LogOutRequest LogOut(const RequestOptions& request_options) override;
+
+  SendAnimationRequest SendAnimation(
+      const SendAnimationMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+  
+  SendAudioRequest SendAudio(
+      const SendAudioMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+  
+  SendChatActionRequest SendChatAction(
+      const SendChatActionMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendContactRequest SendContact(
+      const SendContactMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendDiceRequest SendDice(
+      const SendDiceMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+  
+  SendDocumentRequest SendDocument(
+      const SendDocumentMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendLocationRequest SendLocation(
+      const SendLocationMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendMessageRequest SendMessage(
+      const SendMessageMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendPhotoRequest SendPhoto(
+      const SendPhotoMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendPollRequest SendPoll(
+      const SendPollMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendVenueRequest SendVenue(
+      const SendVenueMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+  
+  SendVideoNoteRequest SendVideoNote(
+      const SendVideoNoteMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendVideoRequest SendVideo(
+      const SendVideoMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+  SendVoiceRequest SendVoice(
+      const SendVoiceMethod::Parameters& parameters,
+      const RequestOptions& request_options) override;
+
+private:
+  template <typename Request>
+  Request FormRequest(
```

Comment:
FormRequest -> CreateRequest
and make it public. You'll got the same result, but less code.